### PR TITLE
Workflow for OSD Version Increment

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Add workflow for OpenSearch Dashboard Plugins Version Increment.
This Workflow uses OSD [plugin-helpers](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/packages/osd-plugin-helpers) `version`.

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/3333
Sample PR: https://github.com/prudhvigodithi/alerting-dashboards-plugin/pull/1/files
Sample Workflow RUN: https://github.com/prudhvigodithi/opensearch-build/actions/runs/4868420202/jobs/8681889881

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/osd-increment-plugin-versions.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+name: Increment OpenSearch Dashboard Plugins Version
+
+on:
```
Comment (gaiksaya) [inline .github/workflows/osd-increment-plugin-versions.yml:4]: Please add cron too:
```
 schedule:
   - cron: 0 0 * * *
```

Comment (prudhvigodithi) [reply .github/workflows/osd-increment-plugin-versions.yml:4]: Sayali initially once merged wanted to run the entire workflow with `workflow_dispatch` before we add the `cron`daily.

**Review by gaiksaya — APPROVED** [review]

File: .github/workflows/osd-increment-plugin-versions.yml (hunk 1 above)
Comment (gaiksaya) [reply .github/workflows/osd-increment-plugin-versions.yml:4]: Sounds good! Wondering if we there is a better way to get all plugins' list from somewhere. Either manifest or https://github.com/opensearch-project/opensearch-plugins

Comment (prudhvigodithi) [reply .github/workflows/osd-increment-plugin-versions.yml:4]: Sure Sayali, let me take a look at it as next step once merged, If I can get that list for `.meta` file and convert to a repo matrix.

File: .github/workflows/osd-increment-plugin-versions.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          - {repo: security-analytics-dashboards-plugin}
+          - {repo: security-dashboards-plugin}
+          - {repo: dashboards-search-relevance}
+        branch:
+          - '1.3'
+          - 2.x
+          - '2.7'
+          - main
+    steps:
+      - name: Check out OpenSearch Dashboard repo
```
Comment (kavilla) [inline .github/workflows/osd-increment-plugin-versions.yml:42]: nit: `OpenSearch Dashboards`

File: .github/workflows/osd-increment-plugin-versions.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
+---
+name: Increment OpenSearch Dashboard Plugins Version
```
Comment (kavilla) [inline .github/workflows/osd-increment-plugin-versions.yml:2]: nit: `OpenSearch Dashboards`

**Review by kavilla — APPROVED** [review]

File: .github/workflows/osd-increment-plugin-versions.yml (hunk 3 above)
Comment (prudhvigodithi) [reply .github/workflows/osd-increment-plugin-versions.yml]: Thanks updated

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**